### PR TITLE
feat: printer interface and migration of all commands

### LINKED ISSUE #201: introduce an interface abstraction for rendering tables with a unified setup
### Describe the feature

at the moment we are using a tablewriter instance created for each command; this makes it really difficult to perform things such as:

- moving to a different library
- implement standard formatting
- support different rendering formats a part from tables

I propose defining a `IPrinter` interface that is provided with the `CliOptions` struct.

Such interface allows commands to define key properties and data rows and invoke a print function. The format to be used is defined by the clioptions.

The interface would be like:

```go
type IPrinter interface {
	Keys(...string) IPrinter
	Record(...string) IPrinter
	BulkRecords(...[]string) IPrinter
	Print()
}
```

All methods are meant to return the same IPrinter instance in order to allow developers chaining method invocations

> [!NOTE]
> The first Printer implementation will be a table printer wrapping the current table implementation

### Code of Conduct

- [X] I agree to follow this project’s Code of Conduct

### PR DESCRIPTION
**Pull request by fredmaggiowski**
## What this PR is for?

This PR addresses #201 proposing the printer interface provided by the `clioptions` package, in order to provide the means for a centralized and standardized render method

**Which issue(s) this PR fixes:**

- Fixes #201 

The Printer interface follows the spec from issue #201 but I've also added basic options management, specifically for wrap lines which have been disabled for certain commands

### CODE REVIEW

File: internal/cmd/marketplace/apply/table.go [hunk 1]
```diff
@@ -19,28 +19,22 @@ import (
...
-	table := tablewriter.NewWriter(strBuilder)
-	table.SetBorder(false)
-	table.SetHeaderAlignment(tablewriter.ALIGN_LEFT)
-	table.SetCenterSeparator("")
-	table.SetColumnSeparator("")
-	table.SetRowSeparator("")
-	table.SetAlignment(tablewriter.ALIGN_LEFT)
-	table.SetAutoWrapText(false)
-	table.SetHeader(headers)
+	// FIXME: should use the printer from clioptions!
```
Comment (fredmaggiowski) [inline internal/cmd/marketplace/apply/table.go:27]: This has been a pain to try and migrate. I've put a FIXME for a future PR

File: internal/cmd/marketplace/list.go [hunk 2]
```diff
@@ -78,14 +75,14 @@ type GetMarketplaceItemsOptions struct {
 	public    bool
 }
 
-func getMarketplaceItemsTable(context context.Context, client *client.APIClient, options GetMarketplaceItemsOptions) (string, error) {
+func getMarketplaceItemsTable(context context.Context, client *client.APIClient, options GetMarketplaceItemsOptions, p printer.IPrinter) error {
```
Comment (malta895) [inline internal/cmd/marketplace/list.go:78]: I would now rename this to `printMarketplaceItemsTable`

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +15,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - update sync to v0.7.0
 - update text to v0.15.0
 
+### Refactor
```
Comment (JGiola) [inline CHANGELOG.md:18]:
```suggestion
### Changed
```

**Review by JGiola — CHANGES_REQUESTED** [review]
Can you also rebase the branch with main?

**Review by JGiola — APPROVED** [review]

**Review by JGiola — APPROVED** [review]

**Pull request merged**